File: src/HttPlaceholder.Models/RequestResultModel.cs
using System;
using System.Collections.Generic;

namespace HttPlaceholder.Models
{
   public class RequestResultModel
   {
      public string CorrelationId { get; set; }

      public object RequestParameters { get; set; }

      public IList<StubExecutionResultModel> StubExecutionResults { get; set; } = new List<StubExecutionResultModel>();

      public string ExecutingStubId { get; set; }

      public DateTime RequestBeginTime { get; set; }

      public DateTime RequestEndTime { get; set; }
   }
}

File: gui/src/domain/request/request-result-model.ts
import type { RequestParametersModel } from "@/domain/request/request-parameters-model";
import type { StubExecutionResultModel } from "@/domain/request/stub-execution-result-model";
import type { StubResponseWriterResultModel } from "@/domain/request/stub-response-writer-result-model";

export interface RequestResultModel {
  correlationId: string;
  requestParameters: RequestParametersModel;
  stubExecutionResults: StubExecutionResultModel[];
  stubResponseWriterResults: StubResponseWriterResultModel[];
  executingStubId: string;
  stubTenant: string;
  requestBeginTime: string;
  requestEndTime: string;
  hasResponse: boolean;
}

export function getDefaultRequestResultModel(): RequestResultModel {
  return {
    requestParameters: {
      body: "",
      url: "",
      clientIp: "",
      headers: {},
      method: "",
    },
    requestEndTime: "",
    requestBeginTime: "",
    correlationId: "",
    executingStubId: "",
    stubExecutionResults: [],
    stubResponseWriterResults: [],
    stubTenant: "",
    hasResponse: false,
  };
}
